test/range.civet

```diff
@@ -56,15 +56,73 @@ describe "range", ->
     ---
     [0...256]
     ---
-    ((s) => Array.from({length: 256 - s}, (_, i) => s + i))(0)
+    var range: (start: number, end: number) => number[] = (start, end) => {
+      const length = end - start;
+      if (length <= 0) return [];
+      const arr = Array(length);
+      for (let i = 0; i < length; ++i) {
+        arr[i] = i + start;
+      }
+      return arr;
+    };
+    range(0,256)
   """
 
+  closedAscendingArr := [0..100]
+  halfClosedAscendingArr := [0...100]
+  openAscendingArr := [1..<100]
```

Comment:
I wonder about making these a little more explicit, so they don't depend on correctness of range literals in the current language:

```js
closedAscendingArr := i for let i = 0; i <= 100; i++ // [0..100]
halfClosedAscendingArr := i for let i = 0; i < 100; i++ // [0...100]
openAscendingArr := i for let i = 1; i < 100; i++ // [1...100]
```

Also, need `it` wrapper.

Reply:
updated